### PR TITLE
Add decoding of av3c - AVS3 configuration box

### PR DESCRIPTION
**Pull request by paulhiggs**
This update adds support for the av3c configuration box used when AVS3 coded video (T/AI 109.2) is transported in an ISO BMFF according to T/AI 109.6:2022 - see http://standard.avswg.org.cn/avs3_download/en_index.asp for specification dosnloads

### CODE REVIEW

File: test/boxHtmlTable.js [hunk 1]
```diff
@@ -42,7 +42,10 @@ function generateBoxTable(box, excluded_fields, additional_props, no_header) {
 					if (i%4 === 3) html += ' ';
 				}
 			} else {
-				html += box[prop];
+				if (box[prop].hasOwnProperty("toString") && typeof box[prop].toString === "function")
+					html += box[prop].toString();
+				else
+					html += box[prop];
```
Comment (cconcolato) [inline test/boxHtmlTable.js:48]: This change does not seem related to AVS3. It may be better in a separate PR linking to an issue reporting a problem.

File: src/parsing/av3C.js [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+				var zero_count=1;
+				while (this.peekBit() == 0) {
+					this.getBit();
+					zero_count++;
+				}
+				var tmp_value = this.getBits(zero_count+1);
+				return tmp_value - 1;
+			}
+		}
+	};
```
Comment (cconcolato) [inline src/parsing/av3C.js:305]: Why not reuse the existing code that reads bits/bytes?

File: src/parsing/av3C.js [hunk 3]
```diff
@@ -0,0 +1,546 @@
+/*
+ * Copyright (c) 2023. Paul Higgs
+ * License: BSD-3-Clause (see LICENSE file)
+ */
+
+var OSisLittleEndian=false;
```
Comment (cconcolato) [inline src/parsing/av3C.js:6]: Global variables should be avoided. 
Also, why would AVS3 need endianness considerations compared to other codecs?

File: src/parsing/av3C.js (hunk 2 above)
Comment (paulhiggs) [reply src/parsing/av3C.js:305]: From what I could see, the existing code only read byte aligned values - the AVS3 sequence header is just a bitstream.
This bit reader is based on the bitstream reader from TSduck.

File: src/parsing/av3C.js (hunk 3 above)
Comment (paulhiggs) [reply src/parsing/av3C.js:6]: The bit reader can return values either as big endian of little endian (useful with native code).
I can fake this out on the assumption  that all will be little endian, of just put it in the class.

File: test/boxHtmlTable.js (hunk 1 above)
Comment (paulhiggs) [reply test/boxHtmlTable.js:48]: Indeed, it is not related to AVS3 but could be used widely if you don't want to display a primative type.
Happy to create a seperate issue and process that way.

**Comment by rbouqueau** [comment]
@paulhiggs Is it expected that the head repository was deleted? (leading to the closing of this PR)

**Comment by paulhiggs** [comment]
Yes because of the concerns raised. I will try to refractor it as an issue and remove the reusability of the endianness.
